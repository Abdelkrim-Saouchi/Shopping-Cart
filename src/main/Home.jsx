// https://www.freepik.com/free-photo/cheerful-satisfied-customer-holding-shopping-bags_8132438.htm
import { Link } from 'react-router-dom';
import heroImg from '../assets/cheerful-satisfied-customer-holding-shopping-bags.jpg';
import styles from './Home.module.css';

const Home = () => {
  return (
    <main className={'container ' + styles.Home}>
      <div className={styles.heroSectionText}>
        <h2>
          Welcome to FutureStore - Your One-Stop Destination for Fashion and
          Technology!
        </h2>
        <p>
          At FutureStore, we believe that shopping should be a delightful
          experience, and that's why we strive to provide you with the best
          products, exceptional customer service, and a seamless shopping
          experience. Browse our diverse range of products, find what you love,
          and let us take care of the rest.
        </p>
        <button>
          <Link to="/products">Explore Now</Link>
        </button>
      </div>
      <img src={heroImg} alt="hero section img" />
    </main>
  );
};

export default Home;
