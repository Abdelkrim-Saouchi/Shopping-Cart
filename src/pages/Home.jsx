// https://www.freepik.com/free-photo/cheerful-satisfied-customer-holding-shopping-bags_8132438.htm
import { Link } from 'react-router-dom';
import heroImg1178 from '../assets/cheerful-satisfied-customer-holding-shopping-bags,w_1178.jpg';
import heroImg1534 from '../assets/cheerful-satisfied-customer-holding-shopping-bags,w_1534.jpg';
import heroImg320 from '../assets/cheerful-satisfied-customer-holding-shopping-bags,w_320.jpg';
import styles from './Home.module.css';

const Home = () => {
  return (
    <main className={'container ' + styles.Home}>
      <div className={styles.heroSectionText}>
        <h2>
          Welcome to FutureStore - Your One-Stop Destination for Fashion and
          Technology!
        </h2>
        <p>
          At FutureStore, we believe that shopping should be a delightful
          experience, and that's why we strive to provide you with the best
          products, exceptional customer service, and a seamless shopping
          experience. Browse our diverse range of products, find what you love,
          and let us take care of the rest.
        </p>
        <Link to="/products">Explore Now</Link>
      </div>
      <img
        sizes="(max-width: 1534px) 100vw, 1534px"
        srcSet={`${heroImg320} 320w, ${heroImg1178} 1178w, ${heroImg1534} 1534w`}
        src={heroImg1534}
        alt=""
      />
    </main>
  );
};

export default Home;
