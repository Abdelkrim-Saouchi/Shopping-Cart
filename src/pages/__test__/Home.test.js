import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { BrowserRouter } from 'react-router-dom';
import Home from '../Home';

test('renders homepage correctly', () => {
  const { container } = render(<Home />, { wrapper: BrowserRouter });
  expect(container).toMatchSnapshot();
});
