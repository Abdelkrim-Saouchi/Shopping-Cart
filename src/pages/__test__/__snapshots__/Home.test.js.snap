// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage correctly 1`] = `
<div>
  <main
    class="container Home"
  >
    <div
      class="heroSectionText"
    >
      <h2>
        Welcome to FutureStore - Your One-Stop Destination for Fashion and Technology!
      </h2>
      <p>
        At FutureStore, we believe that shopping should be a delightful experience, and that's why we strive to provide you with the best products, exceptional customer service, and a seamless shopping experience. Browse our diverse range of products, find what you love, and let us take care of the rest.
      </p>
      <a
        href="/products"
      >
        Explore Now
      </a>
    </div>
    <img
      alt=""
      sizes="(max-width: 1534px) 100vw, 1534px"
      src="cheerful-satisfied-customer-holding-shopping-bags,w_1534.jpg"
      srcset="cheerful-satisfied-customer-holding-shopping-bags,w_320.jpg 320w, cheerful-satisfied-customer-holding-shopping-bags,w_1178.jpg 1178w, cheerful-satisfied-customer-holding-shopping-bags,w_1534.jpg 1534w"
    />
  </main>
</div>
`;
